pkl-core/src/main/java/org/pkl/core/settings/PklSettings.java

```diff
@@ -72,47 +70,40 @@ public static PklSettings load(ModuleSource moduleSource) throws VmEvalException
             .addResourceReader(ResourceReaders.environmentVariable())
             .addEnvironmentVariables(System.getenv())
             .build()) {
-      var module = evaluator.evaluateOutputValueAs(moduleSource, PClassInfo.Settings);
-      return parseSettings(module);
+      var module = evaluator.evaluate(moduleSource);
+      return parseSettings(module, moduleSource);
     }
   }
 
-  private static PklSettings parseSettings(PObject module) throws VmEvalException {
-    // can't use object mapping in pkl-core, so map manually
-    var editor = (PObject) module.getProperty("editor");
-    var urlScheme = (String) editor.getProperty("urlScheme");
-    return new PklSettings(new Editor(urlScheme));
+  private static PklSettings parseSettings(PModule module, ModuleSource location)
+      throws VmEvalException {
+
+    if (!(module.getPropertyOrNull("editor") instanceof PObject pObject)
+        || !(pObject.getPropertyOrNull("urlScheme") instanceof String str)) {
+      throw new VmExceptionBuilder().evalError("invalidSettingsFile", location.getUri()).build();
+    }
+    var editor = new Editor(str);
+    var httpSettings = PklEvaluatorSettings.Http.parse((Value) module.getProperty("http"));
+    return new PklSettings(editor, httpSettings);
   }
 
-  /** Returns the editor for viewing and editing Pkl files. */
+  /**
+   * Returns the editor for viewing and editing Pkl files.
+   *
+   * <p>This method is deprecated, use {@link #editor()} instead.
+   */
+  @Deprecated(forRemoval = true)
   public Editor getEditor() {
     return editor;
   }
 
-  @Override
-  public boolean equals(@Nullable Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
-
-    var that = (PklSettings) o;
-
-    return editor.equals(that.editor);
-  }
-
-  @Override
-  public int hashCode() {
-    return editor.hashCode();
-  }
-
   @Override
   public String toString() {
-    return "PklSettings{" + "editor=" + editor + '}';
+    return "PklSettings{" + "editor=" + editor + ";http=" + http + '}';
```

Comment:
nit: Java will generate a `toString` for a record, similar to this. So not really needed.